lib/image-stream/archive.js

```diff
@@ -81,8 +80,20 @@ const extractEntryByPath = (archive, filePath, options) => {
     return Bluebird.resolve(options.default);
   }
 
-  return options.hooks.extractFile(archive, options.entries, fileEntry.name)
-    .then(rindle.extract);
+  return options.hooks.extractFile(archive, options.entries, fileEntry.name).then((stream) => {
+    return new Bluebird((resolve, reject) => {
+      let data = '';
```

Comment:
Something along the lines of only being able to use this method for "small" files, as it won't work for trying to extract files bigger than the computer's free memory.